framework/packages/abstract-client/src/account.rs

```diff
@@ -145,26 +140,87 @@ impl<'a, Chain: CwEnv> AccountBuilder<'a, Chain> {
         self
     }
 
+    /// Retrieve the account from an installed App.
+    ///
+    /// Queries the app to retrieve the account that the app is installed on.
+    pub fn from_app(&mut self, app: Addr) -> &mut Self {
+        self.fetch_from = Some(AccountSource::App(app));
+        self
+    }
+
+    /// Retrieve the account from an [`AccountId`].
+    pub fn from_account_id(&mut self, account_id: AccountId) -> &mut Self {
+        self.fetch_from = Some(AccountSource::AccountId(account_id));
+        self
+    }
+
+    /// Retrieve the account from a registered [`Namespace`].
+    ///
+    /// Is similar to the `namespace` method, but does not claim the namespace if it is not already claimed.
+    pub fn from_namespace(&mut self, namespace: Namespace) -> &mut Self {
+        self.fetch_from = Some(AccountSource::Namespace(namespace));
+        self
+    }
```

Comment:
It feels odd to have fetch inside builder, since you won't be able to change any other information about account. I would prefer those method on `Account` itself with `TryFrom<AccountSource>`. What do you think?

Reply:
> since you won't be able to change any other information about account.

*I mean we can change it, but we don't want to. Do we want to have `update_account` on the account itself?

Reply:
I think tryfrom accountsource could be a bit weird discover. 

I think we can add it but have the `from_xyz` methods on the `Account` itself instead of the builder, like you suggest.

Reply:
And yeah we can create an update_account that returns a new "builder" based on the old account and just changes the settings but that's for another day imo.